sdk/compute/Azure.ResourceManager.Compute/src/Generated/Models/UserInitiatedRedeploy.cs

```diff
@@ -0,0 +1,69 @@
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+
+namespace Azure.ResourceManager.Compute.Models
+{
+    /// <summary> Specifies Redeploy related Scheduled Event related configurations. </summary>
+    public partial class UserInitiatedRedeploy
+    {
+        /// <summary>
+        /// Keeps track of any properties unknown to the library.
+        /// <para>
+        /// To assign an object to the value of this property use <see cref="BinaryData.FromObjectAsJson{T}(T, System.Text.Json.JsonSerializerOptions?)"/>.
+        /// </para>
+        /// <para>
+        /// To assign an already formatted json string to this property use <see cref="BinaryData.FromString(string)"/>.
+        /// </para>
+        /// <para>
+        /// Examples:
+        /// <list type="bullet">
+        /// <item>
+        /// <term>BinaryData.FromObjectAsJson("foo")</term>
+        /// <description>Creates a payload of "foo".</description>
+        /// </item>
+        /// <item>
+        /// <term>BinaryData.FromString("\"foo\"")</term>
+        /// <description>Creates a payload of "foo".</description>
+        /// </item>
+        /// <item>
+        /// <term>BinaryData.FromObjectAsJson(new { key = "value" })</term>
+        /// <description>Creates a payload of { "key": "value" }.</description>
+        /// </item>
+        /// <item>
+        /// <term>BinaryData.FromString("{\"key\": \"value\"}")</term>
+        /// <description>Creates a payload of { "key": "value" }.</description>
+        /// </item>
+        /// </list>
+        /// </para>
+        /// </summary>
+        private IDictionary<string, BinaryData> _serializedAdditionalRawData;
+
+        /// <summary> Initializes a new instance of <see cref="UserInitiatedRedeploy"/>. </summary>
+        public UserInitiatedRedeploy()
+        {
+        }
+
+        /// <summary> Initializes a new instance of <see cref="UserInitiatedRedeploy"/>. </summary>
+        /// <param name="automaticallyApprove"> Specifies Redeploy Scheduled Event related configurations. </param>
+        /// <param name="dummyProperty"> This is a dummy property to prevent flattening. </param>
+        /// <param name="serializedAdditionalRawData"> Keeps track of any properties unknown to the library. </param>
+        internal UserInitiatedRedeploy(bool? automaticallyApprove, string dummyProperty, IDictionary<string, BinaryData> serializedAdditionalRawData)
+        {
+            AutomaticallyApprove = automaticallyApprove;
+            DummyProperty = dummyProperty;
+            _serializedAdditionalRawData = serializedAdditionalRawData;
+        }
+
+        /// <summary> Specifies Redeploy Scheduled Event related configurations. </summary>
+        public bool? AutomaticallyApprove { get; set; }
+        /// <summary> This is a dummy property to prevent flattening. </summary>
+        public string DummyProperty { get; set; }
```

Comment:
Please add a partial class of this type `UserInitiatedRedeploy` in `src/Custom/Models` or `src/Customization/Models` or similar (see which one we have in this directory) with the following content:
```csharp
namespace Azure.ResourceManager.Compute.Models
{
    public partial class UserInitiatedRedeploy
    {
	private string DummyProperty {get;}
    }
}
```
we cannot use attributes to remove the property because the generator is not updated to properly handle the removed properties yet and removing it will lead to compilation error and needs more customization.
The above customization code only changes the dummy property to private so that it will not be exposed in public API.
Let us try this to see if we could get the expected result.

Reply:
@ArcturusZhang that did not work as I got the error message:
C:\azure-sdk-for-net\sdk\compute\Azure.ResourceManager.Compute\src\Generated\Models\UserInitiatedRedeploy.cs(67,23): er
ror CS0102: The type 'UserInitiatedRedeploy' already contains a definition for 'DummyProperty' [C:\azure-sdk-for-net\sd
k\compute\Azure.ResourceManager.Compute\src\Azure.ResourceManager.Compute.csproj::TargetFramework=netstandard2.0]

Reply:
After generating the code again, when I run export-api.ps1 I see the error:
C:\azure-sdk-for-net\sdk\compute\Azure.ResourceManager.Compute\src\Generated\Models\UserInitiatedRedeploy.Serialization
.cs(35,36): error CS0618: 'UserInitiatedRedeploy.DummyProperty' is obsolete: 'This is a dummy property to prevent flatt
ening.' [C:\azure-sdk-for-net\sdk\compute\Azure.ResourceManager.Compute\src\Azure.ResourceManager.Compute.csproj::Targe
tFramework=netstandard2.0]

How do I customize UserInitiatedRedeploy.Serialization? I'm trying to use the same strategy as UserInitiatedRedeploy.cs to remove Dummy property from UserInitiatedRedeploy.Serialization but it's not working.

Reply:
There is a way of customizing the serialization methods, but I do not recommend you to do that.
The reason is that by customizing the serialization method, you take over the serialization of this model entirely. Next time when we update the swagger, the serialization of this model will not change because it has been taken over manually, and changes must be introduced manually as well. This is very likely to introduce more bugs, for example, new properties are added to the model, but the serialization code does not include them therefore they will be ignored by the requests/responses.
Since this extra property does not break anything in serialization, I suggest we just let it be there

Reply:
Looks like I did get past the initial errors, here is another error unrelated to previous issue that if you could help me understand:

Error		MembersMustExist : Member 'public Azure.Pageable<Azure.ResourceManager.Compute.CapacityReservationGroupResource> Azure.ResourceManager.Compute.ComputeExtensions.GetCapacityReservationGroups(Azure.ResourceManager.Resources.SubscriptionResource, System.Nullable<Azure.ResourceManager.Compute.Models.CapacityReservationGroupGetExpand>, System.Threading.CancellationToken)' does not exist in the implementation but it does exist in the contract.